Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java

```diff
@@ -0,0 +1,120 @@
+package mage.cards.y;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.common.DealCombatDamageControlledTriggeredAbility;
+import mage.abilities.common.DiesCreatureTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.continuous.BecomesFaceDownCreatureEffect;
+import mage.abilities.effects.common.continuous.GainAbilityControlledEffect;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.Card;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SetTargetPointer;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.TargetController;
+import mage.constants.Zone;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.card.FaceDownPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+
+/**
+ *
+ * @author DominionSpy
+ */
+public final class YarusRoarOfTheOldGods extends CardImpl {
+
+    private static final FilterCreaturePermanent filter =
+            new FilterCreaturePermanent("face-down creature");
+
+    static {
+        filter.add(TargetController.YOU.getControllerPredicate());
+        filter.add(FaceDownPredicate.instance);
+    }
+
+    public YarusRoarOfTheOldGods(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{R}{G}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.CENTAUR);
+        this.subtype.add(SubType.DRUID);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Other creatures you control have haste.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityControlledEffect(
+                HasteAbility.getInstance(), Duration.WhileOnBattlefield,
+                StaticFilters.FILTER_PERMANENT_CREATURES, true)));
+        // Whenever one or more face-down creatures you control deal combat damage to a player, draw a card.
+        this.addAbility(new DealCombatDamageControlledTriggeredAbility(
+                Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(1),
+                filter, SetTargetPointer.PLAYER, false)
```

Comment:
should be no need to set target pointer here? simpler constructor should work fine?